test/reflection.jl

```diff
@@ -1036,6 +1036,8 @@ ambig_effects_test(a, b) = 1
     @test Base.infer_effects(===, (Any,Any)) |> Core.Compiler.is_foldable_nothrow
     @test (Base.infer_effects(setfield!, ()); true) # `builtin_effects` shouldn't throw on empty `argtypes`
     @test (Base.infer_effects(Core.Intrinsics.arraylen, ()); true) # `intrinsic_effects` shouldn't throw on empty `argtypes`
+
+    @test Base.@infer_effects(issue41694(1.0)) |> !Core.Compiler.is_nothrow
```

Comment:
It is not in base (tests should go in InteractiveUtils)

Note that this functionality should probably mostly be used via Cthulhu, but I am approving this only because it is tiny anyways to add here.